boards/nordic/nrf54l15pdk/Kconfig

```diff
@@ -0,0 +1,13 @@
+# nRF54L15 PDK board configuration
+
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_NRF54L15PDK_NRF54L15_CPUAPP
+
+config BOARD_ENABLE_DCDC
+	bool "DCDC mode"
+	select SOC_NRF54L_VREG_MAIN_DCDC
+	default y
+
+endif # BOARD_NRF54L15PDK_NRF54L15_CPUAPP
```

Comment:
Think we need to have a discussion with @tejlmand about this now we have hwmv2, doesn't really make sense to introduce a duplicate Kconfig per board when it can just be applied at the SoC level

Reply:
could this be merged as-is and then all boards could be refactored in a separate PR ?

Current implementation is aligned to what we have for nRF52840 DK and nRF5340 DK

Reply:
I don't see why it should go in with this. Changes needed to go to a SoC-level Kconfig for nrf52840dk and nrf5340dk: https://github.com/zephyrproject-rtos/zephyr/pull/70088 (will add other boards later on)

Reply:
Acceptable for now in this PR, will not accept for any boards going forward from this point